include/secp256k1_schnorr_adaptor.h

```diff
@@ -0,0 +1,138 @@
+ *  Args:    ctx: pointer to a context object (not secp256k1_context_static).
+ *  Out:   sig65: pointer to a 65-byte array to store the serialized signature.
+ *  In:    msg32: the 32-byte message being signed.
+ *       keypair: pointer to an initialized keypair.
+ *           t33: pointer to a 33-byte compressed adaptor point.
+ *    aux_rand32: 32 bytes of fresh randomness. While recommended to provide
+ *                this, it is only supplemental to security and can be NULL. A
+ *                NULL argument is treated the same as an all-zero one. See
+ *                BIP-340 "Default Signing" for a full explanation of this
+ *                argument and for guidance if randomness is expensive.
+ */
+SECP256K1_API int secp256k1_schnorr_adaptor_presign(
+    const secp256k1_context *ctx,
+    unsigned char *sig65,
+    const unsigned char *msg32,
+    const secp256k1_keypair *keypair,
+    const unsigned char *t33,
+    const unsigned char *aux_rand32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(5);
+
+/** Extract an adaptor point T from the signature.
+ *
+ *  Returns 1 on success, 0 on failure.
+ *  Args:    ctx: pointer to a context object.
+ *  Out:      t33: pointer to a 33-byte array to store the compressed adaptor point.
+ *  In:     sig65: pointer to a 65-byte serialized signature.
+ *           msg32: the 32-byte message being signed.
+ *           pubkey: pointer to an x-only public key to verify with (cannot be NULL)
+ */
+SECP256K1_API int secp256k1_schnorr_adaptor_extract_t(
+    const secp256k1_context *ctx,
+    unsigned char *t33,
+    const unsigned char *sig65,
+    const unsigned char *msg32,
+    const secp256k1_xonly_pubkey *pubkey
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(5);
+
+/** Adapt an adaptor signature to a schnorr signature.
+ *
+ *  Returns 1 on success, 0 on failure.
+ *  Args:    ctx: pointer to a context object.
+ *  Out:   sig64: pointer to a 64-byte array to store the adapted schnorr signature.
+ *  In:    sig65: pointer to a 65-byte serialized adaptor signature.
+ *           t32: pointer to a 32-byte adaptor.
+ */
+SECP256K1_API int secp256k1_schnorr_adaptor_adapt(
+    const secp256k1_context *ctx,
+    unsigned char *sig64,
+    const unsigned char *sig65,
+    const unsigned char *t32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4);
+
+/** Extract the adaptor from an adaptor signature and a schnorr signature.
+ *
+ *  Returns 1 on success, 0 on failure.
+ *  Args:    ctx: pointer to a context object.
+ *  Out:      t32: pointer to a 32-byte array to store the adaptor.
+ *  In:    sig65: pointer to a 65-byte serialized adaptor signature.
+ *           sig64: pointer to a 64-byte adapted schnorr signature.
+ */
+SECP256K1_API int secp256k1_schnorr_adaptor_extract_adaptor(
```

Comment:
I think we should rename this to `secp256k1_schnorr_adaptor_extract` for simplicity.